misk-api/src/main/kotlin/misk/api/RequestContext.kt

```diff
@@ -0,0 +1,16 @@
+package misk.api
+
+import misk.web.DispatchMechanism
+import okhttp3.HttpUrl
+
+interface RequestContext {
+  val url: HttpUrl
+  val dispatchMechanism: DispatchMechanism
```

Comment:
Afaict, The `method` field in `DispatchMechanism` is the only thing used from DispatchMechanism across different interceptors in cash app repositories. I wonder I should replace this interface value with just a request method. 

I find `DispatchMechanism` a confusing concept given it captures methods or grpc/websocket.

Reply:
I would prefer to keep `DispatchMechanism` still here, as it still could be useful. I would find just having `method` to be confusing as well with grpc

Reply:
A nicer design from my perspective would be to have: 

```
sealed interface DispatchMechanism { // with a different name, so that it's not a braking change
  val method: Method
}

object Grpc : DispatchMechanism {
   override val method: Method = POST
}

object WebSocket : DispatchMechanism {
   override val method: Method = GET
}

data class Http(override val methd: Method): DispatchMechanism

enum Metod {
  // No grpc or websocket here
  GET,
  POST,
  PATCH,
  PUT,
  DELETE
}
```

We can make this change in a non-breaking change if we want. But I'm happy to come back to it later.

Even then though, I think having direct access to `method` in `RequestContext` is nice. Request method is a first-class concern for any http engine.